components/browser/menu/src/test/java/mozilla/components/browser/menu/view/ExpandableLayoutTest.kt

```diff
@@ -0,0 +1,567 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.menu.view
+
+import android.animation.ValueAnimator
+import android.graphics.Rect
+import android.view.MotionEvent
+import android.view.MotionEvent.ACTION_CANCEL
+import android.view.MotionEvent.ACTION_DOWN
+import android.view.MotionEvent.ACTION_MOVE
+import android.view.MotionEvent.ACTION_UP
+import android.view.View
+import android.view.ViewConfiguration
+import android.view.ViewGroup
+import android.view.animation.AccelerateDecelerateInterpolator
+import android.widget.FrameLayout
+import androidx.core.view.marginBottom
+import androidx.core.view.marginLeft
+import androidx.core.view.marginRight
+import androidx.core.view.marginTop
+import androidx.core.view.updateLayoutParams
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import mozilla.components.support.test.any
+import mozilla.components.support.test.mock
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertSame
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers
+import org.mockito.ArgumentMatchers.anyInt
+import org.mockito.Mockito.doAnswer
+import org.mockito.Mockito.doReturn
+import org.mockito.Mockito.never
+import org.mockito.Mockito.spy
+import org.mockito.Mockito.verify
+
+@RunWith(AndroidJUnit4::class)
+class ExpandableLayoutTest {
+    @Test
+    fun `wrapContentInExpandableView should properly setup a new ExpandableLayout`() {
```

Comment:
Maybe we could use [GivenWhenThen](https://martinfowler.com/bliki/GivenWhenThen.html) naming convention to have self document tests and help others to maintain them in the future.

Reply:
Sure. Will try to.
Personally I find it a bit hard or superfluous to write the "Given" in normal unit tests and would prefer to have this naming only in BDD where PMs/BAs can describe the features first using Gherkin and the developers could then go through red-green sessions for implementing those specs.

Reply:
Other than a cool
> e: org.jetbrains.kotlin.backend.common.BackendException: Backend Internal error: Exception during file facade code generation
File being compiled: file:///.../components/browser/menu/src/test/java/mozilla/components/browser/menu/view/ExpandableLayoutTest.kt
The root cause java.io.FileNotFoundException was thrown at: java.io.FileOutputStream.open0(Native Method)
> 
> Caused by: java.io.FileNotFoundException: .../components/browser/menu/build/tmp/kotlin-classes/debugUnitTest/mozilla/components/browser/menu/view/ExpandableLayoutTest$GIVEN parentHeight not calculated WHEN getOrCalculateExpandedHeight is called with a MeasureSpec height size THEN it sets the expandedHeight as the minimum value between expandedHeight and the MeasureSpec height size$expandableLayout$1.class (File name too long)

exception all tests in the `ExpandableLayoutTest` class should follow this convention.
Haven't renamed the ones from `BrowserMenuTest` since the already existent ones don't use this naming style. 
Should I?

Reply:
Don't worry we can keep them as they are. If the naming convention gets to painful we can skip it.